client/jetpack-cloud/sections/settings/advanced-credentials/index.tsx

```diff
@@ -214,9 +214,18 @@ const AdvancedCredentials: FunctionComponent< Props > = ( { action, host, role }
 		if ( formHasErrors ) {
 			return;
 		}
+
+		const credentials = { role, ...formState };
+
+		if ( formMode === FormMode.Password ) {
```

Comment:
The check is done before submitting the credentials rather than when updating `formState`, so that you can switch mode and not loose the value of the password/private key.